app/queries/placements/placements_query.rb

```diff
@@ -30,12 +45,6 @@ def school_type_condition(scope)
     scope.where(schools: { type_of_establishment: params[:school_types] })
   end
 
-  def partner_school_condition(scope)
-    return scope if params[:partner_school_ids].blank?
-
-    scope.where(school_id: params[:partner_school_ids])
-  end
-
```

Comment:
Add this back in and make it more like this:

```ruby
def partner_school_condition(scope)
  return scope unless params[:only_partner_schools]

  provider = params[:context][:current_provider]
  scope.where(school_id: provider.partner_schools.select(:id))
end
```

Changes made:

1. Param name is now `only_partner_schools` and it's a `boolean` so we don't need to check it for 'emptiness' as if it were an array. We only care about whether it's 'truthy' or 'falsey'.
2. `params[:context][:current_provider]` – this is a `Placements::Provider` object, passed in from the controller so we know who the current provider is
3. Using `.select(:id)` instead of `.ids` means this becomes a subquery rather than an additional round-trip to the database. This should be more performant and mean we don't end up holding school IDs in memory.